Tests/Command/CleanMediaCommandTest.php

```diff
@@ -276,8 +288,39 @@ public function testExecute()
 
         $output = $this->tester->execute(array('command' => $this->command->getName()));
 
-        $this->assertRegExp('@Context: foo\s+\'qwertz.ext\' was successfully removed\s+\'thumb_1_bar.ext\' was successfully removed\s+done!@', $this->tester->getDisplay());
-
+        $this->assertOutputFoundInContext(
+            '/Context: foo\s+(.+)\s+done!/ms',
+            array(
+                '\'qwertz.ext\' was successfully removed',
+                '\'thumb_1_bar.ext\' was successfully removed',
+            ),
+            $this->tester->getDisplay()
+        );
         $this->assertSame(0, $output);
     }
+
+    /**
+     * Asserts whether all expected texts can be found in the output within a given context.
+     *
+     * @param string $extract  PCRE regex expected to have a single matching group, extracting the content of a context
+     * @param array  $expected Excerpts of text expected to be found in the output
+     * @param string $output   Output to find
+     */
+    private function assertOutputFoundInContext($extractor, $expected, $output)
+    {
+        preg_match_all($extractor, $output, $matches);
+
+        $found = false;
+        foreach ($matches[1] as $match) {
+            foreach ($expected as $excerpt) {
+                if (false === strpos($match, $excerpt)) {
+                    continue 2;
```

Comment:
> but in this case continue 2; is the same like break; isn't it?

Hmmm yes it might be, since there is nothing after the inner foreach. @lemoinem , can you check?